include/deal.II/base/types.h

```diff
@@ -41,6 +41,12 @@ namespace types
    */
   typedef unsigned int subdomain_id;
 
+  /**
+   * The type used for global indices of vertices.
+   */
+  typedef unsigned long long int global_vertex_index;
+#  define DEAL_II_VERTEX_INDEX_MPI_TYPE MPI_UNSIGNED_LONG_LONG
```

Comment:
I definitely see the need for 64 bit vertex indices on meshes with more than 2^32 cells. So I definitely think that 64 bit is the way to go. The only thing I'd like to discuss is whether we should introduce a new global typedef or if we rather use global_dof_index (although I agree that it would be a slight misnomer in this context). Any other opinions?